### PR TITLE
reapply type-detection logic during 'finish'

### LINKED ISSUE #361: isVideo not properly set for some files
I don't really understand this bug from looking at the source, but i just confirmed it on [files.veliov.com](https://files.veliov.com/) and the `demo-simplest-upload`:  
`isVideo` is not properly set with some video types, specifically matroska videos (`mkv`, at least thats how I noticed it).

You can download my test file from your demo (https://files.veliov.com/QCsG25yCS3eaKARuo) or from the page I originally got it: [mkv at jell.yfish.us](http://jell.yfish.us/media/jellyfish-3-mbps-hd-h264.mkv)

If I log fileRef in `onAfterUpload` I see

    isVideo: false,
    type: 'video/x-matroska',

which should satisfy [the check in line 1242](https://github.com/VeliovGroup/Meteor-Files/blob/master/files.coffee#L1242):

```coffee
      isVideo: /^video\//i.test data.type
      isAudio: /^audio\//i.test data.type
      isImage: /^image\//i.test data.type
      isText:  /^text\//i.test data.type
      isJSON:  /application\/json/i.test data.type
      isPDF:   /application\/pdf|application\/x-pdf/i.test data.type
```

Or is the mimeType that `isVideo` is checked against the one sent by the client, which is later replaced by a `file-type` ([_getMimeType()](https://github.com/VeliovGroup/Meteor-Files/blob/master/files.coffee#L1139))?
If so, shouldn't the `is*` be updated at that time, too?

---

also by the way, why not

```coffee
      isVideo: data.type.startsWith('video/')
      isAudio: data.type.startsWith('audio/')
      isImage: data.type.startsWith('image/')
      isText:  data.type.startsWith('text/')
      isJSON:  data.type === 'application/json'
      isPDF:   data.type === 'application/pdf' || data.type === 'application/x-pdf'
// or at least
      isPDF:   /^application\/(-x)?pdf$/i.test data.type
```

I would assume that to be more performant and I think it reads better.

### PR DESCRIPTION
**Pull request by s-ol**
Close #361.

Turned out to be even easier than I thought 😀

### CODE REVIEW

**Comment by s-ol** [comment]
noticed I failed with the indentation, that's fixed now.

File: files.coffee [hunk 1]
```diff
@@ -1219,6 +1219,22 @@ class FilesCollection
...
+  @returns {Object}
+  ###
+  _updateFileTypes: (data) ->
+    data.isVideo  = /^video\//i.test data.type
+    data.isAudio  = /^audio\//i.test data.type
+    data.isImage  = /^image\//i.test data.type
+    data.isText   = /^text\//i.test data.type
+    data.isJSON   = /^application\/json$/i.test data.type
+    data.isPDF    = /^application\/(x-)?pdf$/i.test data.type
+
```
Comment (dr-dimitru) [inline files.coffee]: In CoffeeScript we need to return something from every function or leave empty `return`. Sorry for CS, v2.0 will be ES6

File: files.coffee [hunk 2]
```diff
@@ -1207,7 +1208,6 @@ class FilesCollection
   @name _getExt
   @param {String} FileName - File name
   @summary Get extension from FileName
-  @returns {Object}
```
Comment (s-ol) [inline files.coffee]: oh, i slipped up again. i wanted to remove the `@returns` from the new function

File: files.coffee (hunk 1 above)
Comment (s-ol) [inline files.coffee]: @dr-dimitru should I return data for chaining or undefined?

**Comment by s-ol** [comment]
@dr-dimitru just realized that dev branch thing too, I guess i need to reopen for this to work on github?

File: files.coffee (hunk 1 above)
Comment (dr-dimitru) [inline files.coffee]: No, just close function with `return` on 1234. It's one of the CS disadvantages

File: files.coffee (hunk 2 above)
Comment (dr-dimitru) [inline files.coffee]: Could you bring it back with one more commit?

File: files.coffee (hunk 1 above)
Comment (s-ol) [inline files.coffee]: @dr-dimitru did that

**Pull request merged**